mlflow/utils/search_utils.py

```diff
@@ -89,8 +91,20 @@ class SearchUtils:
     }
 
     @classmethod
-    def get_sql_filter_ops(cls, column, operator):
-        sql_filter_ops = {"LIKE": column.like, "ILIKE": column.ilike}
+    def get_sql_filter_ops(cls, column, operator, dialect):
+        if dialect == MYSQL:
+
+            def like_op(value):
+                # Filter by `LIKE` ahead of `LIKE BINARY` for runtime performance
+                return sa.and_(
+                    column.like(value), column.op("LIKE BINARY", is_comparison=True)(value)
+                )
+
+        elif dialect == MSSQL:
+            like_op = column.collate("Japanese_Bushu_Kakusu_100_CS_AS_KS_WS").like
```

Comment:
```suggestion
            like_op = column.collate("Japanese_Bushu_Kakusu_100_CS_AS_KS_WS_UTF8").like
```

I tried `Japanese_Bushu_Kakusu_100_CS_AS_KS_WS_UTF8` on local but got an invalid collation error.